src/contentScript.ts

```diff
@@ -15,114 +15,79 @@
 import "@/contentScript/uipath";
-import { isConnectionError } from "@/errors";
 
-const start = Date.now();
+const PIXIEBRIX_SYMBOL = Symbol.for("pixiebrix-content-script");
+const uuid = uuidv4();
+
+declare global {
+  interface Window {
+    [PIXIEBRIX_SYMBOL]?: string;
+  }
+}
 
 async function init(): Promise<void> {
+  // Add error listeners first so they can catch any initialization errors
+  addErrorListeners();
+  addContentScriptListener();
+  addExternalListener();
+  addExecutorListener();
+  initTelemetry();
+
   const sender = await whoAmI();
-  try {
-    updateTabInfo({ tabId: sender.tab.id, frameId: sender.frameId });
-    console.debug(
-      // Safe because we're using our own symbol defined above
-      // eslint-disable-next-line security/detect-object-injection
-      `Loading contentScript for tabId=${sender.tab.id}, frameId=${sender.frameId}: ${window[PIXIEBRIX_SYMBOL]}`
-    );
-  } catch (reason) {
-    console.warn("Error getting tabId/frameId", reason);
-    throw reason;
-  }
 
-  // Refreshing remote services on every page load is too slow
-  // .then(() => {
-  //   // Reload services on background page for each new page. This is inefficient right now, but will
-  //   // avoid confusion if service configurations are updated remotely
-  //   return refreshServices().catch((reason) => {
-  //     console.warn("Error refreshing service configurations", reason);
-  //     throw reason;
-  //   });
-  // })
+  updateTabInfo({ tabId: sender.tab.id, frameId: sender.frameId });
+  console.debug(
+    `Loading contentScript for tabId=${sender.tab.id}, frameId=${sender.frameId}: ${uuid}`
+  );
 
   try {
     await handleNavigate();
-  } catch (reason) {
-    console.warn("Error initializing content script", reason);
-    throw reason;
+  } catch (error) {
+    console.error("Error initializing contentScript", error);
+    throw error;
   }
 
   try {
-    // await the background script know we're ready to execute remote actions
+    // notify the background script know we're ready to execute remote actions
     markReady();
-    console.info(`contentScript ready in ${Date.now() - start}ms`);
     await notifyReady();
-  } catch (reason) {
-    console.warn("Error pinging the background script", reason);
-    throw reason;
+    console.info(`contentScript ready in ${Date.now() - start}ms`);
+  } catch (error) {
+    console.error("Error pinging the background script", error);
+    throw error;
   }
 }
 
-init();
-initTelemetry();
+// Make sure we don't install the content script multiple times
+// eslint-disable-next-line security/detect-object-injection -- using PIXIEBRIX_SYMBOL
+const existing: string = window[PIXIEBRIX_SYMBOL];
+if (!existing) {
+  // eslint-disable-next-line security/detect-object-injection -- using PIXIEBRIX_SYMBOL
+  window[PIXIEBRIX_SYMBOL] = uuid;
+  void init();
+} else {
+  console.debug(`PixieBrix contentScript already installed: ${existing}`);
+}
```

Comment:
Instead of `if not, else`, it's best to use `if, else`

https://eslint.org/docs/rules/no-negated-condition

```suggestion
if (existing) {
  console.debug(`PixieBrix contentScript already installed: ${existing}`);
} else {
  // eslint-disable-next-line security/detect-object-injection -- using PIXIEBRIX_SYMBOL
  window[PIXIEBRIX_SYMBOL] = uuid;
  void init();
}
```